From 94b92246ab29bd20add4ac96c4c88a9c65255b9e Mon Sep 17 00:00:00 2001
From: Daniel Bermond <danielbermond@gmail.com>
Date: Fri, 31 Jul 2020 20:22:16 +0000
Subject: [PATCH] Do not require root access for the build command

Commit bd13868dcb1d9d6b89cc60edcd5a7a022bce516e broke the build
command when running without root access. Building as a non-root
user is useful in some cases and is described in the man page as
a supported feature.

See the discussion in #134

Fixes #134
---
 dkms | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dkms b/dkms
index cf1b608..b0ba3eb 100644
--- a/dkms
+++ b/dkms
@@ -3886,7 +3886,8 @@ for action_to_run in $action; do
         check_root && make_driver_disk
         ;;
     build)
-        check_root && check_all_is_banned "build" && build_modules
+        [ ! -w "$dkms_tree" ] && die 1 $"No write access to create a DKMS tree at ${dkms_tree}"
+        check_all_is_banned "build" && build_modules
         ;;
     add)
         check_root && check_all_is_banned "add" && add_module
